code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm

```diff
@@ -431,19 +431,31 @@ Difficulty: Hard
 		..(force_grant = stored_nearby)
 
 /mob/living/simple_animal/hostile/megafauna/hierophant/devour(mob/living/L)
-	for(var/obj/item/W in L)
-		if(!L.dropItemToGround(W))
-			qdel(W)
+	if(!L || HAS_TRAIT(L, TRAIT_GUTTED))
+		return FALSE
 	visible_message(span_hierophant_warning("\"[pick(kill_phrases)]\""))
-	visible_message(span_hierophant_warning("[src] annihilates [L]!"),span_userdanger("You annihilate [L], restoring your health!"))
+	visible_message(span_hierophant_warning("[src] annihilates [L]'s organs!"),span_userdanger("You annihilate [L]'s organs, restoring your health!"))
 	adjustHealth(-L.maxHealth*0.5)
 	L.investigate_log("has been devoured by [src].", INVESTIGATE_DEATHS)
-	L.dust()
+	var/mob/living/carbon/carbonTarget = L
+	if(istype(carbonTarget))
+		qdel(L.get_organ_slot(ORGAN_SLOT_LUNGS))
+		qdel(L.get_organ_slot(ORGAN_SLOT_HEART))
+		qdel(L.get_organ_slot(ORGAN_SLOT_LIVER))
+	L.adjustFireLoss(500)
+	L.death()
+	RegisterSignal(L, COMSIG_MOB_STATCHANGE, PROC_REF(on_gutted_statchange))
+	ADD_TRAIT(L, TRAIT_GUTTED, src)
+	LoseTarget()
+	return TRUE
```

Comment:
This copy/pasted block is gross (and I know it used to be like that before you touched it).
As far as I can tell, is the only change the text printed to the screen?
Make a new proc called `celebrate_kill` or something and move all the text output into there and override that, allowing the actual functional part of `devour` to remain the same.